src-tauri/src/data_conversion/distance_conversion.rs

```diff
@@ -11,19 +11,28 @@ use crate::mesh::device::MeshNode;
 * Lat/Long: 1e-7 conversion from int to floating point degrees; see mesh.proto
 * Altitude: in meters above sea level, no conversion needed
 */
-pub fn get_distance(node_1: MeshNode, node_2: MeshNode) -> f64 {
-    let node_1_data = node_1.data;
-    let node_2_data = node_2.data;
-    let node_1_pos = node_1_data.position.unwrap();
-    let node_2_pos = node_2_data.position.unwrap();
-    total_distance(
-        node_1_pos.latitude_i as f64 * LAT_CONVERSION_FACTOR,
-        node_1_pos.longitude_i as f64 * LON_CONVERSION_FACTOR,
-        node_1_pos.altitude as f64 * ALT_CONVERSION_FACTOR,
-        node_2_pos.latitude_i as f64 * LAT_CONVERSION_FACTOR,
-        node_2_pos.longitude_i as f64 * LON_CONVERSION_FACTOR,
-        node_2_pos.altitude as f64 * ALT_CONVERSION_FACTOR,
-    )
+pub fn get_distance(node_1: Option<&MeshNode>, node_2: Option<&MeshNode>) -> f64 {
+    let default_distance = 0.0;
+    match(node_1, node_2) {
+        (Some(node_1), Some(node_2)) => {
```

Comment:
Nit: If it makes it easier, you could even do an `if let (node1, node2) = (Some(node1), Some(node2)) { ... return value; } return default ` to replace the `match` statements with `if` statements

Reply:
for some reason, they stay as Options when I do that.

```
if let (node1, node2) = (Some(node1), Some(node2)) {
   let node1_pos = node1.data.position;
   ...
}
```
doesn't let me access the node1.data field because it's an Option.
Leaving it as a match for that reason.